duckduckgo/iOS
# network tracker leaderboard

Reviewer: Mia
Asana: https://app.asana.com/0/414235014887631/457416080701398/f
CC:

**Description**:

Adds the Network Leaderboard screen and the Top 3 Tracking Networks on the initial screen.

Also updated initial screen to reflect networks rather than unique trackers.

Also updated the Javascript to find the parent domain if matched with Easylist (though I am yet to see it actually find a parent domain but I tested the logic via the Javascript console).

**Steps to test this PR**:

All tests are cumulative.

Test 1
1. Visit example.com 
1. Initial screen shows "still gathering data" message
1. Leaderboard screen shows 0%, no trackers, no reset button

Test 2
1. Visit evanscycles.com
1. Initial screen shows "still gathering data" message
1. Leaderboard screen shows 0%, no trackers, no reset button
1. Turn off protection
1. Top 3 appear
1. Leaderboard shows list of networks

Test 3
1. Visit StarWars.com (protection off)
1. While loading show privacy protection - the top 3 updates in real time
1. The leaderboard is also updated in real time

Test 4
1. Visit enough sites to gather enough data so that the reset button "floats" when on the leaderboard
1. Scrolling to the bottom automatically hides the reset button
1. Re-open the leaderboard and dismiss the reset button

Test 5
1. Tap reset
1. The leaderboard is reset as per Test 1
1. The top 3 on the initial screen is reset as per Test 1

Test 6
1. Visit more sites to gather data and confirm that top 3 and leaderboard are repopulated.
